include/aspect/postprocess/geoid.h

```diff
@@ -0,0 +1,347 @@
+
+        /**
+         *Do the multipole expansion at a particular quadrature point.
+         *
+         *@param position The location of the quadrature point.
+         *
+         *@param value  The value of the function being expanded.
+         *
+         *@param weigth The Jacobian and weight given to the quadrature point
+         *
+         */
+        virtual void add_quadrature_point (const Point<dim> &position, const double value, const double weight);
+
+        /**
+         * Evaluate the multipole expansion at point p
+         */
+        virtual double evaluate( const Point<dim> &p ) const;
+
+        /**
+         * Get the maximum degree of the expansion
+         */
+        unsigned int degree() const;
+
+        /**
+         * Return a reference to the internal representation of the mulipole expansion.
+         */
+        const HarmonicCoefficients<dim> &get_coefficients () const;
+
+        /*
+         * Set all the multipole coefficients to zero.
+         */
+        void clear();
+
+        /*
+         * Scalar add another multipole expansion to this one.
+         * Computes this = s*this + a*M for each coefficient.
+         */
+        void sadd( double s, double a, const MultipoleExpansion &M);
+
+        /*
+         * Scalar add another multipole expansion to this one.
+         * Each degree can have a different scalar factor, so
+         * the size of s and a are expected to be max_degree+1.
+         * Computes this = s[l]*this + a[l]*M for each coefficient.
+         */
+        void sadd( const std::vector<double> &s, const std::vector<double> &a, const MultipoleExpansion &M);
+
+
+        /**
+         * Perform an MPI sum on the coefficents.
+         */
+        void mpi_sum_coefficients (MPI_Comm mpi_communicator);
+
+        const unsigned int max_degree;  //expansion degree
+        HarmonicCoefficients<dim> coefficients; //sine and cosine coefficients
+      };
+
+      /**
+       * Multipole expansion for the case where the evaluation point
+       * is external to the sources.
+       */
+      template <int dim>
+      struct ExternalMultipoleExpansion : public MultipoleExpansion<dim>
+      {
+        ExternalMultipoleExpansion ( const unsigned int max_degree, const double evaluation_radius );
+        /**
+         *Do the multipole expansion at a particular quadrature point.
+         *
+         *@param position The location of the quadrature point.
+         *
+         *@param value  The value of the function being expanded.
+         *
+         *@param weigth The Jacobian and weight given to the quadrature point
+         *
+         */
+        virtual void add_quadrature_point (const Point<dim> &position, const double value, const double weight);
+
+        /**
+         * Evaluate the multipole expansion at point p
+         */
+        virtual double evaluate( const Point<dim> &p ) const;
+
+        const double evaluation_radius;
```

Comment:
It is the radius at which we are measuring the gravitational potential, so eather R_E or R_CMB